presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQueryConfig.java

```diff
@@ -0,0 +1,193 @@
+    private int maxReadRowsRetries = DEFAULT_MAX_READ_ROWS_RETRIES;
+
+    @AssertTrue(message = "Either one of 'bigquery.credentials-key' or 'bigquery.credentials-file' must be specified, or the default GoogleCredentials could be created")
+    public boolean isCredentialsConfigurationValid()
+    {
+        // at most one of them should present
+        if (credentialsKey.isPresent() && credentialsFile.isPresent()) {
+            return false;
+        }
+        // if no credentials, try create default ones
+        if (!credentialsKey.isPresent() && !credentialsFile.isPresent()) {
+            try {
+                GoogleCredentials.getApplicationDefault();
+            }
+            catch (IOException e) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Optional<String> getCredentialsKey()
+    {
+        return credentialsKey;
+    }
+
+    @Config("bigquery.credentials-key")
+    @ConfigDescription("credentials key (base64 encoded)")
+    public BigQueryConfig setCredentialsKey(String credentialsKey)
+    {
+        this.credentialsKey = Optional.of(credentialsKey);
+        return this;
+    }
+
+    public Optional<String> getCredentialsFile()
+    {
+        return credentialsFile;
+    }
+
+    @Config("bigquery.credentials-file")
+    @ConfigDescription("JSON credentials file path")
+    public BigQueryConfig setCredentialsFile(String credentialsFile)
+    {
+        this.credentialsFile = Optional.of(credentialsFile);
+        return this;
+    }
+
+    public Optional<String> getProjectId()
+    {
+        return projectId;
+    }
+
+    @Config("bigquery.project-id")
+    @ConfigDescription("Google Cloud project ID")
+    public BigQueryConfig setProjectId(String projectId)
+    {
+        this.projectId = Optional.of(projectId);
+        return this;
+    }
+
+    public Optional<String> getParentProjectId()
+    {
+        return parentProjectId;
+    }
+
+    @Config("bigquery.parent-project-id")
+    @ConfigDescription("Google Cloud parent project ID")
+    public BigQueryConfig setParentProjectId(String parentProjectId)
+    {
+        this.parentProjectId = Optional.of(parentProjectId);
+        return this;
+    }
+
+    public OptionalInt getParallelism()
+    {
+        return parallelism;
+    }
+
+    @Config("bigquery.parallelism")
+    @ConfigDescription("The number of partitions to split the data")
+    public BigQueryConfig setParallelism(int parallelism)
+    {
+        this.parallelism = OptionalInt.of(parallelism);
+        return this;
+    }
+
+    public boolean isViewsEnabled()
+    {
+        return viewsEnabled;
+    }
+
+    @Config("bigquery.views-enabled")
+    @ConfigDescription("Enable BigQuery connector to read views")
+    public BigQueryConfig setViewsEnabled(boolean viewsEnabled)
+    {
+        this.viewsEnabled = viewsEnabled;
+        return this;
+    }
+
+    public int getViewExpirationTimeInHours()
+    {
+        return 24;
```

Comment:
is this defined somewhere in bigquery class files? nice to import the constant from bigquery classes

Reply:
no but I can define a constant variable as a default config value